### ISSUE TITLE
[BUG] avoid-border-all is triggered even when it is not a const

### ISSUE THREAD

**Issue by ValentinVignal** [dart-code-checker/dart-code-metrics#973]
There is a regression which is exactly https://github.com/dart-code-checker/dart-code-metrics/issues/837. The rule `avoid-border-all` is triggered even when the parameters are not constants.

* **Dart code metrics version:**
`4.18.0-dev.2`
* **Dart sdk version:**
`2.17.6`

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yaml
include: package:flutter_lints/flutter.yaml

analyzer:
  plugins:
    - dart_code_metrics

dart_code_metrics:
  rules:
    - avoid-border-all
```

</details>

**What did you do? Please include the source code example causing the issue.**

You can also checkout https://github.com/ValentinVignal/flutter_app_stable/tree/dart-code-metrics/border-and-radius

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    final boolean = MediaQuery.of(context).size.width > 300;
    final color = boolean ? Colors.red : Colors.black;
    return MaterialApp(
      home: Container(
        decoration: BoxDecoration(
          border: Border.all(color: color), // <- Lint
        ),
      ),
    );
  }
}

```

**What did you expect to happen?**

I expect to see no lint when the variables inside the border are not constant

**What actually happened?**

I see a lint

```
Prefer using const constructor Border.fromBorderSide.
const Border.fromBorderSide(BorderSide(color: Theme.of(context).colorScheme.primary)) dart[avoid-border-all](https://dartcodemetrics.dev/docs/rules/flutter/avoid-border-all).
```

**Are you willing to submit a pull request to fix this bug?**

With help, I could

**Comment by incendial** [comment]
Interesting to have a regression, will check

**Comment by incendial** [comment]
Available in 4.18.0 🚀
